### PR TITLE
Update docs for HTTP feature

### PR DESCRIPTION
**Pull request by ierceg**

### CODE REVIEW

**Comment by ianand** [comment]
Task linked: [XDN-5710 Update docs](https://app.clickup.com/t/4205457/XDN-5710)

File: guides/security.md [hunk 1]
```diff
@@ -46,7 +46,28 @@ environment variables will not enforce basic authentication.
 
 ## SSL
 
-The Moovweb XDN only accepts traffic over `https`. It automatically redirects `http` requests to the same URL, including any query strings, on `https`.
+The Moovweb XDN by default only serves the traffic over `https` protocol. It automatically redirects `http` requests to the same URL, including any query strings, on `https`.
```
Comment (markbrocato) [inline guides/security.md:49]:
```suggestion
By default the Moovweb XDN only serves traffic over the `https` protocol. It automatically redirects `http` requests to the same URL, including any query strings, on `https`.
```

File: guides/security.md [hunk 2]
```diff
@@ -46,7 +46,28 @@ environment variables will not enforce basic authentication.
...
+
+We strongly discourage the use of `http` protocol but if you *must* enable it then you can explicitly match on `http` protocol. For example:
```
Comment (markbrocato) [inline guides/security.md:51]:
```suggestion
We strongly discourage the use of `http` protocol but if you *must* enable it then you can do so by adding `protocol: 'http'` to your route criteria. For example:
```

File: guides/security.md [hunk 3]
```diff
@@ -46,7 +46,28 @@ environment variables will not enforce basic authentication.
...
+// Respond to Let's Encrypt HTTP-01 challenge.
+router.match({
+  protocol: 'http',
+  path: '/.well-known/acme-challenge/<your token>'
+}, ({ send }) => {
+  send('<token value>')
+})
+```
+
+If no route is matched on `http` protocol then the XDN will fallback on its default behavior of automatically redirecting the traffic to `https`.
```
Comment (markbrocato) [inline guides/security.md:65]:
```suggestion
If no route is matched on `http` protocol then the XDN will fallback on its default behavior of automatically redirecting the request to `https`.
```

File: guides/security.md [hunk 4]
```diff
@@ -46,7 +46,28 @@ environment variables will not enforce basic authentication.
...
+
+Additionally:
+
+* The received protocol can be read from [`x-xdn-protocol`](request_headers#section_general_headers) request header or from [`req.secure`](/docs/api/core/interfaces/_router_request_.request.html#secure) flag in serverless.
```
Comment (markbrocato) [inline guides/security.md:69]:
```suggestion
* A request's protocol can be determined by reading the [`x-xdn-protocol`](request_headers#section_general_headers) request header or the [`request.secure`](/docs/api/core/interfaces/_router_request_.request.html#secure) property.
```

File: guides/security.md [hunk 5]
```diff
@@ -46,7 +46,28 @@ environment variables will not enforce basic authentication.
...
+* Running in local development assumes that you are developing on `https` protocol. When you want to test your router for `http` protocol matching you should either set `xdn_emulate_local_http` cookie to `true` (if using a browser) or send `x-xdn-protocol` request header set to `http`.
```
Comment (markbrocato) [inline guides/security.md:70]:
```suggestion
* During local development all requests will appear secure by default.  To test your router for `http` protocol matching you must either set the `xdn_emulate_local_http` cookie to `true` (if using a browser) or send an `x-xdn-protocol` request header set to `http`.
```

**Review by markbrocato — CHANGES_REQUESTED** [review]
See my suggestions.

**Review by markbrocato — APPROVED** [review]

**Pull request merged**